tests/app/parser/test_metadata_parser.py

```diff
@@ -177,3 +177,31 @@ def test_business_params_without_schema_name(fake_business_metadata_runner):
     claims = validate_runner_claims(fake_business_metadata_runner)
 
     assert claims["schema_name"] == "mbs_0253"
+
+
+def test_when_response_id_is_missing(fake_business_metadata_runner):
+    expected = (
+        f"{fake_business_metadata_runner['ru_ref']}"
+        f"{fake_business_metadata_runner['collection_exercise_sid']}"
+        f"{fake_business_metadata_runner['eq_id']}"
+        f"{fake_business_metadata_runner['form_type']}"
+    )
+    del fake_business_metadata_runner["response_id"]
```

Comment:
We should do a version of this where it is set to a falsely value like `""`, upstream have done this in the past and I guess is why Mark asked us to guard key not existing or empty value.